### PR TITLE
Exo35

### PR DESCRIPTION
**Pull request by tianyuchen**

### CODE REVIEW

File: 035-Circular_primes.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    else:
+        # only odd numbers
+        for i in range(3, int(n ** .5) + 1, 2):
+            if n % i == 0:
+                return False
+    return True
+
+
+def rotation(n):
+    N = []
```
Comment (gameboy1024) [inline 035-Circular_primes.py]: Use meaningful var names. N doesn't mean anything to someone who looks at this code

File: 035-Circular_primes.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    rot = []
```
Comment (gameboy1024) [inline 035-Circular_primes.py]: Having short names like "rot", nb, qrt, lmd has negative impact on readability of the code. Please refactor to sth more meaningful. I.e. rotations

File: 035-Circular_primes.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        N.append(i)
+
+    # all rotations of number of separate digits
+    for i in range(len(N)):
+        # ex. N[i:] = [1, 9, 7], [9, 7], [7]  N[:i] = [], [1], [1, 9]
+        rot.append(N[i:] + N[:i])
+
+    # join separate digits to number
+    for num in rot:
+        res.append(int(''.join(str(digit) for digit in num)))
```
Comment (gameboy1024) [inline 035-Circular_primes.py]: Check again what's the type of "digit" in num.

File: 035-Circular_primes.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    return res
+
+
+def circular_primes(n):
+    total = 0
+    for i in range(2, 10 ** n):
+        for num in rotation(i):
```
Comment (gameboy1024) [inline 035-Circular_primes.py]: https://www.programiz.com/python-programming/methods/built-in/any

File: 035-Circular_primes.py (hunk 1 above)
Comment (tianyuchen) [reply 035-Circular_primes.py]: changed

File: 035-Circular_primes.py (hunk 2 above)
Comment (tianyuchen) [reply 035-Circular_primes.py]: t'as raison

File: 035-Circular_primes.py (hunk 4 above)
Comment (tianyuchen) [reply 035-Circular_primes.py]: it should use all(), changed !

Comment (gameboy1024) [reply 035-Circular_primes.py]: any() === !all()

File: 035-Circular_primes.py [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+        list.append(i)
+
+    # all rotations of number of separate digits
+    for i in range(len(list)):
+        # ex. N[i:] = [1, 9, 7], [9, 7], [7]  N[:i] = [], [1], [1, 9]
+        separate_rotations.append(list[i:] + list[:i])
+
+    # join separate digits to number
+    for num in separate_rotations:
+        rotations.append(int(''.join(digit for digit in num)))
```
[the remark this answers is not in the document]
Comment (tianyuchen) [reply 035-Circular_primes.py]: num 是['1', '9', '7'] 我需要把它变成197啊

**Pull request merged**